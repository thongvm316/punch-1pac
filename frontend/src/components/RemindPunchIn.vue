<template>
  <div
    v-if="currentUser.forgot_punch_in_days_in_month.length > 0 && open"
    class="toast toast-error text-center mt-2"
  >
    <button
      ref="btnCloseToast"
      class="btn btn-clear float-right"
      @click="close"
    />
    {{ $t('remind.message', { days: formatDays }) }}
  </div>
</template>

<script>
import { mapState, mapMutations } from 'vuex'
import { UPDATE_REMIND_PUNCH_IN } from '@/store/mutation-types'

export default {
  name: 'RemindPunchIn',

  computed: {
    ...mapState('remindPunchIn', ['open']),

    formatDays() {
      let forgotDays = this.currentUser.forgot_punch_in_days_in_month
      return forgotDays.length ? forgotDays.map(day => this.$moment(day).format('L')).join(' , ') : null
    }
  },

  methods: {
    ...mapMutations('remindPunchIn', [UPDATE_REMIND_PUNCH_IN]),

    close() {
      this[UPDATE_REMIND_PUNCH_IN](false)
    }
  }
}
</script>
