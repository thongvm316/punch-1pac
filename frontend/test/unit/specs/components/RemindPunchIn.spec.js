import { shallowMount } from '@vue/test-utils'

import wrapperOps from '../../supports/wrapper'

import RemindPunchIn from '@/components/RemindPunchIn'

const close = jest.spyOn(RemindPunchIn.methods, 'close')

describe('RemindPunchIn.vue', () => {
  let wrapper

  afterEach(() => { wrapper.vm.$destroy() })

  describe('when user remember to punch in all day', () => {
    beforeEach(() => {
      wrapper = shallowMount(RemindPunchIn, {
        ...wrapperOps,
        computed: {
          currentUser() {
            return { forgot_punch_in_days_in_month: [] }
          }
        }
      })

      it('should render correctly', () => {
        expect(wrapper.exists()).toBeTruthy()
        expect(wrapper.isVueInstance()).toBeTruthy()
      })

      it('should not show the reminder toast', () => {
        expect(wrapper.isVisible()).toBeeFalsy()
      })
    })
  })

  describe('when user have forgot punch in some day', () => {
    beforeEach(() => {
      wrapper = shallowMount(RemindPunchIn, {
        ...wrapperOps,
        computed: {
          currentUser() {
            return { forgot_punch_in_days_in_month: ['2018-05-07', '2018-05-08', '2018-06-20'] }
          }
        },
        methods: {
          close
        }
      })
    })

    it('should show the remind punch in toast', () => {
      expect(wrapper.isVisible()).toBeTruthy()
    })

    it('should call close when click close button', async () => {
      wrapper.find({ ref: 'btnCloseToast' }).trigger('click')
      await wrapper.vm.$nextTick()

      expect(close).toHaveBeenCalled()
    })
  })
})
