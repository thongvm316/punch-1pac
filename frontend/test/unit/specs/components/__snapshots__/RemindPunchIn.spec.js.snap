// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemindPunchIn.vue when user have forgot punch in some day should show the remind punch in toast 1`] = `
<div class="toast toast-error text-center mt-2"><button class="btn btn-clear float-right"></button>
  You forgot to punch in on May 07 , May 08 , Jun 20. Please send request to admin for editing attendances on those days !!!
</div>
`;

exports[`RemindPunchIn.vue when user remember to punch in all day should render correctly 1`] = ``;
